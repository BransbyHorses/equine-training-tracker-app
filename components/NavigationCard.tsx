import React, { ReactNode } from "react";
import { Typography, Grid, Paper, Box, IconButton } from "@mui/material";
import Link from "next/link";
import KeyboardArrowRightIcon from "@mui/icons-material/KeyboardArrowRight";

export interface NavigationCardProps {
	link: string;
	title: string;
	icon?: ReactNode;
}

export default function NavigationCard({
	link,
	title,
	icon,
}: NavigationCardProps) {
	return (
		<Grid item xs={12} sm={6}>
			<Paper>
				<Link href={link}>
					<Box
						px={2}
						py={2}
						sx={{
							display: "flex",
							justifyContent: "space-between",
							alignItems: "center",
							cursor: "pointer",
						}}
					>
						<Typography variant="h6">{title}</Typography>
						<IconButton>
							{icon ? icon : <KeyboardArrowRightIcon fontSize="large" />}
						</IconButton>
					</Box>
				</Link>
			</Paper>
		</Grid>
	);
}
