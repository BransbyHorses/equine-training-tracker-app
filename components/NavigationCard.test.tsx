import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import NavigationCard from "./NavigationCard";
import AddCircleIcon from "@mui/icons-material/AddCircle";

describe("NavigationCard", () => {
	it("renders with minimum props", () => {
		const tree = renderer.create(<NavigationCard link="/" title="Test" />);
		expect(tree).toMatchSnapshot();
	});
	it("renders with all props", () => {
		const tree = renderer.create(
			<NavigationCard link="/" title="Test" icon={<AddCircleIcon />} />
		);
		expect(tree).toMatchSnapshot();
	});
});
