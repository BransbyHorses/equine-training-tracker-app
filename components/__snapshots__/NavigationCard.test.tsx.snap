// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationCard renders with all props 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-13i4rnv-MuiGrid-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
  >
    <div
      className="MuiBox-root css-1k9xkr1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Test
      </h6>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddCircleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`NavigationCard renders with minimum props 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-13i4rnv-MuiGrid-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
  >
    <div
      className="MuiBox-root css-1k9xkr1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Test
      </h6>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
          data-testid="KeyboardArrowRightIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
